### PR TITLE
Fix FreeBSD Interface Exception

### PR DESCRIPTION
**Pull request by constrict**
#250 

Null is returned by a pregmatch not an empty string.

### CODE REVIEW

**Comment by kesselb** [comment]
Are you able to reproduce this on your box? I would like to add unit tests at some point for the network logic (like we already have for memory and disks) so every breaking input is nice to have ;)

I'm tried one of the reports from the issue but the input seems to work: https://3v4l.org/lPiVo

**Comment by constrict** [comment]
I was able to reproduce the issue by adding a bridge interface.

Checking for the null does work. I also added a default for a null status.

image

```
bridge0: flags=8802<BROADCAST,SIMPLEX,MULTICAST> metric 0 mtu 1500
        ether 00:00:00:00:00:00
        id 00:00:00:00:00:00 priority 32768 hellotime 2 fwddelay 15
        maxage 20 holdcnt 6 proto rstp maxaddr 2000 timeout 1200
        root id 00:00:00:00:00:00 priority 0 ifcost 0 port 0
        groups: bridge
        nd6 options=9<PERFORMNUD,IFDISABLED>

```

File: lib/OperatingSystems/FreeBSD.php [hunk 1]
```diff
@@ -184,10 +184,15 @@ public function getNetworkInterfaces() {
 				preg_match("/(?<=\<).*(?=-)/m", $intface, $duplex);
 
 				$iface['mac'] = implode(' ', $mac[0]);
-				$iface['status'] = $status[0];
 				$iface['speed']  = $speed[0];
 
-				if ($iface['speed'] !== '') {
+				if ($status[0] !== null) {
```
Comment (kesselb) [inline lib/OperatingSystems/FreeBSD.php:189]:
```suggestion
				if (isset($status[0])) {
```

**Review by kesselb — APPROVED** [review]

File: lib/OperatingSystems/FreeBSD.php (hunk 1 above)
Comment (constrict) [reply lib/OperatingSystems/FreeBSD.php]: Changed all three statements to isset.

**Pull request merged**